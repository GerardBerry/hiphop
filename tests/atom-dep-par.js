"use hopscript"

const hh = require("hiphop")

const prg =
    <hh.module>
      <hh.outputsignal name="A" combine_with=${(x, y) => x + y}/>

      <hh.parallel>
	<hh.loop>
	  <hh.emit signal_name="A" args=0 />
	  <hh.pause/>
	</hh.loop>

	<hh.loop>
	  <hh.emit signal_name="A" args=1 />
	  <hh.atom func=${x => {console.log(x)}} args=${hh.value("A")}/>
	  <hh.pause/>
	</hh.loop>

	<hh.loop>
	  <hh.emit signal_name="A" args=2 />
	  <hh.atom func=${x => {console.log(x)}} args=${hh.value("A")}/>
	  <hh.pause/>
	</hh.loop>
      </hh.parallel>

    </hh.module>;

let machine = new hh.ReactiveMachine(prg, "error2");

console.log(machine.react())
console.log(machine.react())
console.log(machine.react())
console.log(machine.react())
console.log(machine.react())
