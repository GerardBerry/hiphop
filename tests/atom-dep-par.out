3
3
[ 'A(3)' ]
6
6
[ 'A(6)' ]
9
9
[ 'A(9)' ]
12
12
[ 'A(12)' ]
15
15
[ 'A(15)' ]
